packages/core-data/src/footnotes/index.js

```diff
@@ -56,30 +61,20 @@ export function updateFootnotesFromMeta( blocks, meta ) {
 				continue;
 			}
 
-			// When we store rich text values, this would no longer
-			// require a regex.
-			const regex =
-				/(<sup[^>]+data-fn="([^"]+)"[^>]*><a[^>]*>)[\d*]*<\/a><\/sup>/g;
-
-			attributes[ key ] = value.replace(
-				regex,
-				( match, opening, fnId ) => {
-					const index = newOrder.indexOf( fnId );
-					return `${ opening }${ index + 1 }</a></sup>`;
-				}
-			);
-
-			const compatRegex = /<a[^>]+data-fn="([^"]+)"[^>]*>\*<\/a>/g;
-
-			attributes[ key ] = attributes[ key ].replace(
-				compatRegex,
-				( match, fnId ) => {
-					const index = newOrder.indexOf( fnId );
-					return `<sup data-fn="${ fnId }" class="fn"><a href="#${ fnId }" id="${ fnId }-link">${
-						index + 1
-					}</a></sup>`;
-				}
-			);
+			const richTextValue = create( { html: value } );
+
+			richTextValue.replacements.forEach( ( replacement ) => {
+				if ( replacement.type !== 'core/footnote' ) return;
```

Comment:
😢 no curlies makes security-bot sad

Reply:
Aww

Reply:
Curious as to why it's a security issue?

Reply:
it's a code pattern that's risky because of how easy it is to separate the body from the if statement. it was this code pattern in the famous Apple SSL defect that let in full access through basically any system that relied on SSL (basically everything)

code tools aren't generally good at being aware of these; people tend to look past them; comments are often inserted in the wrong place and can even break these control structures without it being obvious.

this one particular case probably isn't that risky, but it's the habit of doing so that makes security bot sad. the cost/benefit ratio for arguable inconvenience to gaping code defects is out of what with this, just like it is with automatic semicolon insertion and that strange zeal people had for leaving out unnecessary semicolons.